hazenlib/slice_width.py

```diff
@@ -109,18 +118,90 @@ def get_rods(dcm):
 
     rods = [Rod(x=x[1], y=x[0]) for x in rods]
     rods = sort_rods(rods)
+    rods_initial = deepcopy(rods)  # save for later
+
+    """
+    Gaussian 2D Rod Locator
+    """
+
+    # setup bounding box dict
+    bbox = {"x_start": [], "x_end": [], "y_start": [], "y_end": [], "intensity_max": [], "rod_dia": [], "radius": []}
+
+    # get relevant label properties
+    rod_radius = []
+    rod_inv_intensity = []
+
+    rprops = regionprops(labeled_array, arr_inv)[1:]  # ignore first label
+    for idx, i in enumerate(rprops):
+        rod_radius.append(rprops[idx].feret_diameter_max)  # 'radius' of each label
+        rod_inv_intensity.append(rprops[idx].intensity_max)
+
+    rod_radius_mean = int(np.mean(rod_radius))
+    rod_inv_intensity_mean = int(np.mean(rod_inv_intensity))
+    bbox["radius"] = int(np.ceil((rod_radius_mean * 2) / 2))
```

Comment:
could we just convert to int at the end when it needs to be an int, and use the floats while we're working out the box value. which should help minimise rounding errors

Reply:
The first two are purely used for extracting the pixel region of the bounding box, hence they need to be ints. The last line is just to initialise the 2D Gaussian fit. The final estimated rod coordinates are floats rather than ints.

Reply:
Discussed with Laurence – agreed not vital.